colossalai/shardformer/modeling/llama.py

```diff
@@ -141,7 +141,9 @@ def llama_model_forward(
                 invert=(sp_mode != "ring_attn"),
             )
         else:
-            attn_kwargs: torch.Tensor = self._update_causal_mask(attention_mask, hidden_states, cache_position)
+            attn_kwargs: torch.Tensor = self._update_causal_mask(
+                attention_mask, hidden_states, cache_position, past_key_values, output_attentions
+            )
```

Comment:
We don't need this?

Reply:
this can be removed here.
but this is another bug, this did not work when you train llama3, llama3.1, llama3.2

https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/llama/benchmark.py
i hope you can try this, and use HybridParallelPlugin

Reply:
I'm not sure what you refer to, `colossalai run --nproc_per_node 2 --master_port 29501 benchmark.py -p 3d -b 1 -g --zero 2 ` (flash attn disabled, so go into this if branch) doesn't throw any error. To justify such changes and save time, please provide a command to easily reproduce the error.